examples/Madnlp_densekkt_example.jl

```diff
@@ -0,0 +1,87 @@
+using DynamicNLPModels, NLPModels, Random, LinearAlgebra, MadNLP, QuadraticModels, MadNLPGPU, CUDA, SparseArrays
+include("build_thinplate.jl")
+
+
+function MadNLP.jac_dense!(nlp::DenseLQDynamicModel{T, V, M1, M2, M3}, x, jac) where {T, V, M1<: Matrix, M2 <: Matrix, M3 <: Matrix}
+    NLPModels.increment!(nlp, :neval_jac)
+    J = nlp.data.A
+    copyto!(jac, J)
+end
+
+function MadNLP.hess_dense!(nlp::DenseLQDynamicModel{T, V, M1, M2, M3}, x, w1l, hess; obj_weight = 1.0) where {T, V, M1<: Matrix, M2 <: Matrix, M3 <: Matrix}
+    NLPModels.increment!(nlp, :neval_hess)
+    H = nlp.data.H
+    copyto!(hess, H)
+end
+
+function convert_to_CUDA!(lqdm::DenseLQDynamicModel)
+    H = CuArray{Float64}(undef, size(lqdm.data.H))
+    J = CuArray{Float64}(undef, size(lqdm.data.A))
+
+    LinearAlgebra.copyto!(H, lqdm.data.H)
+    LinearAlgebra.copyto!(J, lqdm.data.A)
+
+    lqdm.data.H = H
```

Comment:
shouldn't modify the model, but should recreate a model with concrete type parameters

Reply:
True. In general, I prefer to not use `mutable` structure if we can